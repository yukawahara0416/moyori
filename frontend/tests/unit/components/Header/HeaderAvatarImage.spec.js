import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import Component from '@/components/Header/HeaderAvatarImage.vue'

const localVue = createLocalVue()
localVue.use(Vuex)

let wrapper
let propsData
let store

beforeEach(() => {
  propsData = {
    currentUser: {
      data: {
        id: 1,
        name: 'test',
        image: 'image'
      }
    }
  }

  wrapper = shallowMount(Component, {
    localVue,
    propsData,
    store
  })
})

afterEach(() => {
  wrapper.destroy()
})

describe('props', () => {
  it('currentUser', () => {
    expect(wrapper.props().currentUser).toStrictEqual(propsData.currentUser)
    expect(wrapper.props().currentUser instanceof Object).toBe(true)
  })
})

describe('template', () => {
  it('snapshot', () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
