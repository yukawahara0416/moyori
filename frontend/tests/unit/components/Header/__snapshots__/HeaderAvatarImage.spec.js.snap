// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template snapshot 1`] = `
<v-avatar-stub
  class="white--text headline"
  color="success"
  size="36"
>
  <img
    alt=""
    src=""
  />
</v-avatar-stub>
`;
